src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,123 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exaption.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    @Override
+    public Book create(Book book) {
+        String sql = "INSERT INTO books (title, price) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new RuntimeException("Expected to insert at list one row, but insert 0 rows");
+            }
+
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add data to table", e);
+        }
+        return book;
```

Comment:
Move `return` inside try block
```suggestion
            }
            return book;
        } catch (SQLException e) {
            throw new DataProcessingException("Can't add data to table", e);
        }
       
```

Reply:
not fixed